Fundamentals

okhaimie-dev opened the pull request:
<!-- Description of PR... -->

## Changes

- Created Fundamentals Page

https://github.com/Developer-DAO/academy-turbo/issues/28

Markkos89 commented:
gm @okhaimie-dev 
can you add any sort of descriptive text to the PR? the screenshot works but you can add plenty of information, notes, comments, things to take a look at, tech debt, info that you needed and is missing already, and maybe more
can you update the title of the PR to be more descriptive? What means `Fundamentals` only? Take a look at #33 as an example

thanks

Markkos89 commented on apps/academy/src/pages/_app.tsx:
```diff
@@ -100,8 +101,44 @@ type AppPropsWithLayout<P> = AppProps<P> & {
+  const router = useRouter();
   const getLayout = Component.getLayout ?? ((page) => page);
 
+  // If the current route is '/fundamentals', don't use the Layout
+  if (router.pathname === "/fundamentals") {
+    return (
+      <WagmiConfig config={wagmiConfig}>
+        <SessionProvider refetchInterval={0} session={pageProps.session}>
+          <RainbowKitSiweNextAuthProvider>
+            <RainbowKitProvider
+              chains={chains}
+              initialChain={polygonMumbai}
+              appInfo={{
+                appName: "Developer DAO Academy",
+                learnMoreUrl: "https://academy.developerdao.com",
+              }}
+              theme={{
+                lightMode: academyLightTheme,
+                darkMode: academyDarkTheme,
+              }}
+            >
+              <ThemeProvider attribute="class">
+                <DefaultSeo {...SEO} />
+                <Head>
+                  <meta
+                    name="viewport"
+                    content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no, viewport-fit=cover"
+                  />
+                </Head>
+                <Component {...pageProps} />
+              </ThemeProvider>
+            </RainbowKitProvider>
+          </RainbowKitSiweNextAuthProvider>
+        </SessionProvider>
+      </WagmiConfig>
+    );
+  }
```
why don't using the layout? 
The layout includes the connect button and the navigation bar
the layout was made to be used

okhaimie-dev replied:
This page layout is split into two sides. The left side is fixed, while the right side needs to scroll.
![CleanShot 2023-10-27 at 08 16 22](https://github.com/Developer-DAO/academy-turbo/assets/57156589/09833168-947d-4003-a670-96cd6aec9fb7)


The default layout doesn't seem to enable that.

Yes, the layout was made to be used, although in this case, didn't seem like the ideal fit to achieve this result of a fixed left side and a scrolling right side per the design.

@peteruche21 can you chime in here to see if I am missing something?

Take a look at the demo to see what I am referring to:
https://academy-turbo-git-fundamentals-developdao.vercel.app/fundamentals

Markkos89 replied:
> The layout is split into two sides in the fundamental and track page case. The left side is fixed, while the right side needs to scroll. ![CleanShot 2023-10-27 at 08 16 22](https://user-images.githubusercontent.com/57156589/278653738-09833168-947d-4003-a670-96cd6aec9fb7.gif) [ ![CleanShot 2023-10-27 at 08 16 22](https://user-images.githubusercontent.com/57156589/278653738-09833168-947d-4003-a670-96cd6aec9fb7.gif) ](https://user-images.githubusercontent.com/57156589/278653738-09833168-947d-4003-a670-96cd6aec9fb7.gif) [ ](https://user-images.githubusercontent.com/57156589/278653738-09833168-947d-4003-a670-96cd6aec9fb7.gif)
> 
> The default layout doesn't seem to enable that.
> 
> Yes, the layout was made to be used, although in this case, didn't seem like the ideal fit to achieve this result of a fixed left side and a scrolling right side per the design.
> 
> @peteruche21 can you chime in here to see if I am missing something?
> 
> Take a look at the demo to see what I am referring to: https://academy-turbo-git-fundamentals-developdao.vercel.app/fundamentals

I see it, but using the route.path to compare when it's fundamentals page and to not use the layout, is not the way to go. 
a. we are duplicating code
b. we are not generating a layout that can be shareable to other page (homepage and viceversa)

okhaimie-dev replied:
I understand. My rationale was to make something that works first, then work on optimization.

I will abstract the layout from the `fundamentals.tsx` page and move it to `_app.tsx`.

Markkos89 commented on apps/academy/src/pages/fundamentals.tsx:
```diff
@@ -0,0 +1,62 @@
+    <div className="flex h-screen w-full flex-col bg-black lg:flex-row">
+      <div className="flex h-full min-h-screen flex-1 flex-col items-center justify-between bg-[url('/fundamental-bg.jpeg')] bg-cover bg-no-repeat object-center lg:fixed lg:inset-y-0 lg:w-1/2">
+        <div className="mt-10">
+          <TopBar menus={sampleMenus} />
```
use the layout component for this
the Topbar menu is included https://github.com/Developer-DAO/academy-turbo/blob/b0163eed1151a91ee4a14ed1433823a2361f36f2/apps/academy/src/components/Header.tsx#L28

peteruche21 replied:
@okhaimie-dev I think you can still make it work while using the layout. 

I updated the way the layout header is displayed on my last PR #33

Some steps I think could work 
- Position the fundamentals page as grid
- make the height and width take 100% of the viewport
- make the second grid col scrollable and wrap your contents in it


Some trade off being the footer will be visible.

@Markkos89 If possible I think the footer should be decoupled from the layout as only the home page uses it to allow the approach above work.

Markkos89 replied:
> @okhaimie-dev I think you can still make it work while using the layout.
> 
> I updated the way the layout header is displayed on my last PR #33
> 
> Some steps I think could work
> 
> * Position the fundamentals page as grid
> * make the height and width take 100% of the viewport
> * make the second grid col scrollable and wrap your contents in it
> 
> Some trade off being the footer will be visible.
> 
> @Markkos89 If possible I think the footer should be decoupled from the layout as only the home page uses it to allow the approach above work.

Loving this, thanks a lot @peteruche21 !

@okhaimie-dev can follow peter's comments? We will appreciate it.

@peteruche21 regarding the footer not beeing in all pages on the designs, can we include it anyways? It's my opinion, for me makes sense to it be present on all pages, what do you guys think? cc @okhaimie-dev @wolovim @elPiablo

okhaimie-dev replied:
Having the Footer show up in the Fundamental page, will throw off the design done in Figma for Fundamentals and Tracks page.

![CleanShot 2023-10-27 at 17 58 27](https://github.com/Developer-DAO/academy-turbo/assets/57156589/4351c9a6-08a1-4f74-a2bb-f3ad1b4144fb)

So, just to clarify @Markkos89 

1. You are asking me to go off the design done by @uma-l, and proceed to include the Footer to the fundamental page?

This will span across both cols (left & right)? And will I be able to make it stick to the bottom of the left and right cols while still keeping the right col scrollable. = remember the left col is fix = I reckon it will only leave a small view space for the fundamental lessons to the right

![CleanShot 2023-10-27 at 18 14 36](https://github.com/Developer-DAO/academy-turbo/assets/57156589/3316b443-7bee-4fe4-93af-1eb6e83dbedf)

Markkos89 replied:
@okhaimie-dev was my idea but, for the sake of moving forward, please submit the PR without the footer. We can then update the page IF the footer will be included.

Thank you Okhai!!! :rocket: 

Best regards

okhaimie-dev commented:
> gm @okhaimie-dev can you add any sort of descriptive text to the PR? the screenshot works but you can add plenty of information, notes, comments, things to take a look at, tech debt, info that you needed and is missing already, and maybe more can you update the title of the PR to be more descriptive? What means `Fundamentals` only? Take a look at #33 as an example
> 
> thanks

@Markkos89 I am not sure what you mean by this. I did describe that I created the fundamentals page.

> gm @okhaimie-dev can you add any sort of descriptive text to the PR? the screenshot works but you can add plenty of information, notes, comments, things to take a look at, tech debt, info that you needed and is missing already, and maybe more can you update the title of the PR to be more descriptive? What means `Fundamentals` only? Take a look at #33 as an example
> 
> thanks

I renamed it to created fundamentals page.

You can review the page by clicking the "fundamentals" menu item.



> gm @okhaimie-dev can you add any sort of descriptive text to the PR? the screenshot works but you can add plenty of information, notes, comments, things to take a look at, tech debt, info that you needed and is missing already, and maybe more can you update the title of the PR to be more descriptive? What means `Fundamentals` only? Take a look at #33 as an example
> 
> thanks

I edited the PR title and added more context. p

Markkos89 commented on .vscode/settings.json:
```diff
@@ -26,5 +26,8 @@
   "typescript.tsdk": "root/node_modules/typescript/lib",
-  "workbench.editor.labelFormat": "medium"
+  "workbench.editor.labelFormat": "medium",
+  "[typescriptreact]": {
+    "editor.defaultFormatter": "ms-vsliveshare.vsliveshare"
+  }
```
@okhaimie-dev this is your config, can you keep it for you only? Pushing this will change all of us settings.

Thanks

okhaimie-dev replied:
alright. done

Markkos89 replied:
> alright. done

Apologize @okhaimie-dev I didn't mean to delete the whole `.vscode` folder, just deleting the new options added on this PR was enough.
The other should remain as they were before please :pray: 

Thanks

okhaimie-dev replied:
I will look up ways to achieve this in the am

okhaimie-dev replied:
Alright, so the way I can solve this issue is to add `.vscode` folder to the `.gitignore` file after we go back to the initial state. 

I just did that.

Review and let me know if there is anything else. I will get started working on the mobile view.

okhaimie-dev replied:
cc: @Markkos89
